### ISSUE TITLE
How to get the gradients during training?

### ISSUE THREAD

**Issue by RogerXT** [OpenNMT/OpenNMT-py#982]
I am working on doing some modifications for OpenNMT. However, I am still not very familiar with the details of implementation.

After back propagation, how do we get the gradients? 

May I ask for some hints?

**Comment by vince62s** [comment]
I don't know what you are trying to do or understand, but if you want to modify you will have to undertsand the code.
Main training loop is in trainer.py
and gradients are updated here: https://github.com/OpenNMT/OpenNMT-py/blob/master/onmt/trainer.py#L294-L318
Please ask on the forum this kind of questions, thanks.
